karate-core/src/main/java/com/intuit/karate/core/StepRuntime.java

```diff
@@ -173,6 +200,7 @@ public static Result execute(Step step, Actions actions) {
             return Result.failed(0, e, step);
         }
         MethodMatch match = matches.get(0);
+        step.setMatch(match);
```

Comment:
since this doesn't belong on the core `Step` domain, should this be on the `StepResult`

Reply:
if fine either way however to add it to the StepResult domain will probably need to change the following builder (for consistent should always be available):

![image](https://user-images.githubusercontent.com/53574318/105607357-2abfdc00-5d6c-11eb-8095-3149b4393a04.png)

which is used across several places including to describe the ScenarioResult at some times
![image](https://user-images.githubusercontent.com/53574318/105607430-ac176e80-5d6c-11eb-8f13-aa49e569b8ab.png)

I think keeping that inside step might be less impactful although I agree that it should probably belong to the StepResult domain (as a characteristic of the execution, not of the "parsed" test script)

Reply:
although right after posting this noticed your new method Step.fromKarateJson() and in cases where we're reading from JSON this will be null so it'd be inconsistent anyways.

let me push my latest changes while I think about this one

Reply:
if it helps - it is ok to have a non-final variable on `StepResult` even though I try to have everything immutable